presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralEncoder.java

```diff
@@ -197,18 +205,56 @@ public Expression toExpression(Object object, Type type)
             // This if condition will evaluate to true: object instanceof Slice && !type.equals(VARCHAR)
         }
 
+        Signature signature = getMagicLiteralFunctionSignature(type);
         if (object instanceof Slice) {
             // HACK: we need to serialize VARBINARY in a format that can be embedded in an expression to be
             // able to encode it in the plan that gets sent to workers.
             // We do this by transforming the in-memory varbinary into a call to from_base64(<base64-encoded value>)
             FunctionCall fromBase64 = new FunctionCall(QualifiedName.of("from_base64"), ImmutableList.of(new StringLiteral(VarbinaryFunctions.toBase64((Slice) object).toStringUtf8())));
-            Signature signature = FunctionRegistry.getMagicLiteralFunctionSignature(type);
             return new FunctionCall(QualifiedName.of(signature.getName()), ImmutableList.of(fromBase64));
         }
-
-        Signature signature = FunctionRegistry.getMagicLiteralFunctionSignature(type);
-        Expression rawLiteral = toExpression(object, FunctionRegistry.typeForMagicLiteral(type));
+        Expression rawLiteral = toExpression(object, typeForMagicLiteral(type));
 
         return new FunctionCall(QualifiedName.of(signature.getName()), ImmutableList.of(rawLiteral));
     }
+
+    public static boolean isSupportedLiteralType(Type type)
+    {
+        return SUPPORTED_LITERAL_TYPES.contains(type.getJavaType());
+    }
+
+    public static Signature getMagicLiteralFunctionSignature(Type type)
+    {
+        TypeSignature argumentType = typeForMagicLiteral(type).getTypeSignature();
+
+        return new Signature(MAGIC_LITERAL_FUNCTION_PREFIX + type.getTypeSignature(),
```

Comment:
I'd format this like
```java
String name = MAGIC_LITERAL_FUNCTION_PREFIX + type.getTypeSignature();
return new Signature(name, SCALAR, type.getTypeSignature(), argumentType);

Reply:
Normally when I do refactoring I don't change the original code. I found this easier for code review. (You can safely ignore then content when reviewing as long as it still compiles). But if we also want to make the code nicer, I'm open to it. So far I didn't change the name or implementation of logic refactored. Let me know if you think otherwise.